Round robin edits from Josh (#174)
AlexsLemonade/ScPCA-manuscript: content/03.results.md

Reviewer: Actually while I'm recommending looking at L147, there are also TSV file names there for bulk files. Do we want to remove those too from Results?

If the merged object contains an `altExp` with merged ADT data, two `AnnData` objects are exported to create separate RNA (`_rna.h5ad`) and ADT (`_adt.h5ad`) objects.

`merge.nf` outputs a summary report for each merged object, which includes a set of tables summarizing the types of samples and libraries included in the project, such as types of diagnosis, and a faceted UMAP showing all cells from all libraries.
In the faceted plot, each panel highlights a different library included in the merged object, with cells from the specified library shown in color, while all other cells are gray.

> <!-- JAS: I would probably cut the previous sentence entirely; the level of detail seems excessive for the main text -->

Author: I was expecting to leave the sentence below, so no need to move the reference??

<!-- TODO: ⚠️ For review - What do you think of the next sentence? -->
Using this measure, we found that the `BlueprintEncodeData` reference [@doi:10.3324/haematol.2013.094243; @doi:10.1038/nature11247], which includes a variety of normal cell types, tended to perform similarly to or better than other references when applied to samples from different disease types (Figure {@fig:figS4}).
Based on these findings, we used the `BlueprintEncodeData` reference to annotate cells from all libraries on the Portal.
Use of a consistent reference also supports cross-project analyses.

In contrast, `CellAssign` is a marker-gene-based annotation method that requires a binary matrix with all cell types and all associated marker genes as the reference.
We used the list of marker genes available as part of `PanglaoDB` [@doi:10.1093/database/baz046] to construct organ-specific marker gene matrices with marker genes from all cell types listed for the specified organ.
Since many cancers may have infiltrating immune cells, all immune cells were also included in each organ-specific reference.

> For each ScPCA project, we used the organ-specific marker gene matrix relevant to the disease and tissue type from which the sample was obtained (e.g., for brain tumors, we used a brain-specific marker gene matrix with all brain and immune cell types).

Reviewer: I realize don't really like saying "disease" here - since the marker genes are from healthy tissue this feels a bit unclear. 

```suggestion
For each ScPCA project, we used the organ-specific marker gene matrix that most closely matched the tissue type from which the sample was obtained (e.g., for brain tumors, we used a brain-specific marker gene matrix with all brain and immune cell types).
```

If `CellAssign` cannot find a likely cell type from the marker gene matrix, it does not assign a cell type.
Because we annotate cells from tumor samples using references containing only normal cells, we anticipate that many cells, particularly the tumor cells, will not have an exact match; reporting this to the end user is valuable.
Indeed, when applying `CellAssign` to tumor samples with our chosen reference, we observed that many cells were unassigned.

> We included an example in Figure {@fig:figS5}A where unassigned cell types are labeled with `Unknown`.

Reviewer: The phrase "reporting this to the end user is valuable" feels a bit terse. I was working to rephrase this but wondered if we should actually just delete it? I'm not sure this info contributes much more than saying we report these with an `Unknown` label.

When comparing annotations obtained from `CellAssign` and `SingleR` annotations to submitter-provided annotations, we noticed the labels for non-tumor cells were similar between `CellAssign`, `SingleR`, and submitter annotations, while the tumor cells were not assigned using `CellAssign` (Figure {@fig:figS5}B).

### Adding cell type annotations to the ScPCA Portal

`scpca-nf` adds cell type annotations from `SingleR` and `CellAssign` to all processed `SingleCellExperiment` objects (Figure {@fig:fig4}A).
This requires two additional reference files as input to the workflow: a classification model built from a reference dataset for `SingleR` and a marker gene matrix for `CellAssign`.
`SingleR::trainSingleR()` was used to build a classification model from the provided `BlueprintEncodeData` dataset and create the required `SingleR` input for `scpca-nf`.
The classification model and processed `SingleCellExperiment` were input to `SingleR::classifySingleR()`, resulting in annotations for all cells and an associated matrix with per-cell scores for all possible cell type labels.

> The `SingleR` score matrix and the assigned cell types are added to the processed objects output by `scpca-nf`.

Reviewer: ```suggestion
The `SingleR` score matrix and the assigned cell types are added to each processed object output by `scpca-nf`.
```

Simultaneously, processed `SingleCellExperiment` objects are converted to `AnnData` objects for classification with `CellAssign`.
`CellAssign` uses the converted `AnnData` object and the marker gene matrix to train a model and predict the most likely cell type from the possible cell types in the marker gene matrix.
The `CellAssign` prediction matrix, which contains a probability that each cell is one of each possible cell types, and the assigned cell types are added to the processed objects output by `scpca-nf`.

An additional cell type report with information about reference sources, comparisons among cell type annotation methods, and diagnostic plots is also output by `scpca-nf`.
Tables summarizing the number of cells assigned to each cell type for each method are shown alongside UMAPs coloring cells by the assigned cell type.
The concordance of cell type annotations assigned between both methods can indicate higher confidence in the provided annotations.
We therefore used the Jaccard similarity index to compare annotations between the two methods, as well as submitter-provided annotations, if available.

> This index is calculated between pairs of cell type labels from each method and ranges from 0-1, with a value close to 1 indicating high agreement and a high proportion of overlapping cells and values close to 0 indicating a low proportion of overlapping cells.

Reviewer: making it parallel

```suggestion
This index is calculated between pairs of cell type labels from each method and ranges from 0-1, where values close to 1 indicate high agreement and a high proportion of overlapping cells and values close to 0 indicate a low proportion of overlapping cells.
```

We expect cells with the same or similar cell type labels using both methods will likely be more accurately annotated.
`scpca-nf` therefore assigns consensus cell type labels when the two automated methods agree.
To account for different levels of granularity in reference datasets, we employed an ontology-based approach to assign a consensus cell type label.

> Specifically, the consensus cell type annotation is equivalent to the latest common ancestor (LCA) shared between the two predicted cell types.

Reviewer: Confirming this `ontoProc` info is in Methods, so 👍

To ensure specificity in the consensus labels, cells were only assigned a consensus cell type if the identified LCA had no more than 170 descendant terms, with a few exceptions (see Materials and Methods for more details).
This threshold was chosen to exclude overly general cell ontology terms, such as lymphocyte, while retaining meaningful classifications like T cell and B cell.
After assigning all consensus cell types, we looked at the expression of cell-type specific marker genes across all cells to validate the assignments (Figure {@fig:fig5}A, Figure {@fig:figS6}).

> The addition of consensus cell type labels provides a harmonized cell type annotation for all samples in the ScPCA Portal, facilitating downstream analyses across multiple samples.

Reviewer: ```suggestion
The consensus cell type labels provide harmonized cell type annotations for all samples in the ScPCA Portal, facilitating downstream analyses across multiple samples.
```
